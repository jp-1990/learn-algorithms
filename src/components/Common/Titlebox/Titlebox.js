import React from "react"
import PropTypes from "prop-types"

import styles from "./Titlebox.module.scss"

const Titlebox = ({ title, type }) => {
  return (
    <div className={styles.container}>
      <div className={styles.titlebox}>
        <h1 data-testid="title">{title}</h1>
      </div>
      <div className={styles.typebox}>
        <h4 data-testid="type">{type}</h4>
      </div>
    </div>
  )
}

Titlebox.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
}

export default Titlebox
