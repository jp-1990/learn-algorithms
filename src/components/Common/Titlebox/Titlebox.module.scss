.container {
  @include flexCenter(center, center);

  flex-direction: column;
  color: $mainText;
  text-transform: uppercase;
  background-color: $colorA;
  padding: 10px;
  .titlebox {
    @include flexCenter(center, center);
    @include borderRadius;
    background-color: $colorC;
    font-weight: 300;
    width: auto;
    padding: 18px 10px 12px 10px;
  }
  .typebox {
    @include flexCenter(center, center);

    background-color: $colorE;
    width: 110%;
    padding: 8px;
    font-weight: 600;
    letter-spacing: 2px;
  }
}
