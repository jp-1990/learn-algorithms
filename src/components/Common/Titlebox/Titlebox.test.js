import React from "react"
import renderer from "react-test-renderer"
import { render } from "@testing-library/react"

import Titlebox from "./Titlebox"

describe("<Titlebox />", () => {
  const defaultProps = { title: "hello", type: "world" }
  const wrapper = renderer.create(<Titlebox {...defaultProps} />)

  test("render", () => {
    expect(wrapper).toMatchSnapshot()
  })
  test("title", () => {
    const { getByTestId } = render(<Titlebox {...defaultProps} />)
    expect(getByTestId("title")).toHaveTextContent(`${defaultProps.title}`)
  })
  test("type", () => {
    const { getByTestId } = render(<Titlebox {...defaultProps} />)
    expect(getByTestId("type")).toHaveTextContent(`${defaultProps.type}`)
  })
})
