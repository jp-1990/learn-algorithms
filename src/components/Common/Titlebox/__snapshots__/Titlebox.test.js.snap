// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Titlebox /> render 1`] = `
<div
  className="container"
>
  <div
    className="titlebox"
  >
    <h1
      data-testid="title"
    >
      hello
    </h1>
  </div>
  <div
    className="typebox"
  >
    <h4
      data-testid="type"
    >
      world
    </h4>
  </div>
</div>
`;
